packages/sitecore-jss-nextjs/src/components/NextImage.tsx

```diff
@@ -0,0 +1,164 @@
+import { mediaApi } from '@sitecore-jss/sitecore-jss/media';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { getEEMarkup } from '@sitecore-jss/sitecore-jss-react';
+import Image, { ImageLoader, ImageLoaderProps, ImageProps as NextImageProps } from 'next/image';
+export interface ImageFieldValue {
+  [attributeName: string]: unknown;
+  src?: string;
+  /** HTML attributes that will be appended to the rendered <img /> tag. */
+}
+
+export interface ImageField {
+  value?: ImageFieldValue;
+  editable?: string;
+}
+
+export interface ImageSizeParameters {
+  [attr: string]: string | number | undefined;
+  /** Fixed width of the image */
+  w?: number;
+  /** Fixed height of the image */
+  h?: number;
+  /** Max width of the image */
+  mw?: number;
+  /** Max height of the image */
+  mh?: number;
+  /** Ignore aspect ratio */
+  iar?: 1 | 0;
+  /** Allow stretch */
+  as?: 1 | 0;
+  /** Image scale. Defaults to 1.0 */
+  sc?: number;
+}
+
+export interface ImageProps extends Partial<NextImageProps> {
+  [attributeName: string]: unknown;
+  /** Image field data (consistent with other field types) */
+  field?: ImageField | ImageFieldValue;
+
+  /**
+   * Can be used to explicitly disable inline editing.
+   * If true and `media.editable` has a value, then `media.editable` will be processed
+   * and rendered as component output. If false, `media.editable` value will be ignored and not rendered.
+   */
+  editable?: boolean;
+
+  /**
+   * Parameters that will be attached to Sitecore media URLs
+   */
+  imageParams?: {
+    [paramName: string]: string | number;
+  };
+  /**
+   * Custom regexp that finds media URL prefix that will be replaced by `/-/jssmedia` or `/~/jssmedia`.
+   * @example
+   * /\/([-~]{1})assets\//i
+   * /-assets/website -> /-/jssmedia/website
+   * /~assets/website -> /~/jssmedia/website
+   */
+  mediaUrlPrefix?: RegExp;
+
+  /** HTML attributes that will be appended to the rendered <img /> tag. */
+}
+
+export const loader: ImageLoader = ({ src, width }: ImageLoaderProps): string => {
+  const url = new URL(`https://cm.jss.localhost${src}`);
```

Comment:
Need to have a dynamic hostname, it will not work outside container env

Reply:
Ideally, we can utilize the `next.config.js`'s `images` `path` config here, though I thought we were having trouble getting that to work. Were you able to figure that out?

If this is not working, can we leave this as a relative URL (this is where we encountered a bug in Sitecore referenced below, I believe)? 

Otherwise, I'm not sure how we'd grab the proper server URL to use. The `SITECORE_API_HOST` env variable is not guaranteed.

Reply:
According to what i have been reading the `images` `path` does not work with the custom loader. It is only configurable with the build-in loaders provided by next.